frontend/public/module/k8s/network.ts

```diff
@@ -25,48 +17,41 @@ export type ClusterNetworkFeatures = {
   [key in ClusterNetworkFeature]?: boolean;
 };
 
-/**
- * Main document depicting all the features that are supported by each supported CNI type.
- * Undefined features would require to take an ambiguous action (e.g. allow the customer
- * to set the policy Egress rule in a form, but show a warning explaining that this field is
- * not available for the Openshift SDN type)
- */
-const featuresDocument: {
-  [k in ClusterNetworkType]: ClusterNetworkFeatures;
-} = {
-  OpenShiftSDN: {
-    PolicyEgress: false,
-    PolicyPeerIPBlockExceptions: false,
-  },
-  OVNKubernetes: {
-    PolicyEgress: true,
-    PolicyPeerIPBlockExceptions: true,
-  },
-  Unknown: {},
+const getFeatureState = (data: { [key: string]: string }, key: string): boolean | undefined => {
+  // Note: config map data comes as string, not bool
+  return data.hasOwnProperty(key) ? data[key] === 'true' : undefined;
 };
 
 /**
  *  Fetches and returns the features supported by the Cluster Network Type
- *  (Openshift SDN or Kubernetes OVN)
+ *  (Openshift SDN, Kubernetes OVN ...) using a config map provided by the
+ *  cluster network operator.
  *
  *  @async
  *  @returns [ClusterNetworkFeatures, boolean, any] - The asynchronously-loaded cluster network
  *  features, plus a boolean that is 'false' until the first value is loaded (or an error is
  *  returned)
  */
 export const useClusterNetworkFeatures = (): [ClusterNetworkFeatures, boolean] => {
-  const [features, setFeatures] = React.useState<ClusterNetworkFeatures>(
-    featuresDocument[ClusterNetworkType.Unknown],
-  );
+  const [features, setFeatures] = React.useState<ClusterNetworkFeatures>({});
   const [featuresLoaded, setFeaturesLoaded] = React.useState(false);
-  const [network, networkLoaded] = useK8sGet<K8sResourceKind>(clusterNetworkModel, networkName);
+  const [config, configLoaded] = useK8sGet<K8sResourceKind>(
+    ConfigMapModel,
+    networkConfigMapName,
+    networkConfigMapNamespace,
+  );
   React.useEffect(() => {
-    if (networkLoaded) {
-      const cniType = ClusterNetworkType[network?.spec?.networkType];
-      setFeatures(featuresDocument[cniType ?? ClusterNetworkType.Unknown]);
+    if (configLoaded && config?.data) {
+      setFeatures({
+        PolicyEgress: getFeatureState(config.data, policyEgressConfigKey),
```

Comment:
For sure I'm missing some internal TypeScript/JavaScript detail, but: isn't this the same as just writing
```
PolicyEgress: config.data[policyEgressConfigKey],
```
???